src/deploy.ts

```diff
@@ -200,18 +200,29 @@ export const deployPackage = async (
 };
 
 export const deployFromRepository = async (config: Config, plan: string) => {
-	//TODO fetch branch list via api and then show option rather than taking branch name
-	const branch = await input('Enter branch name');
-	//TODO check if repo url is valid
+	const api = API(config.token as string, config.baseURL);
+
+	let branch;
 	const url = args['addrepo'];
 
-	info(`Deploying from ${url as string}...\n`);
+	try {
+		const { branches } = await api.branchList(url as string);
 
-	const api = API(config.token as string, config.baseURL);
+		if (!branches.length) return error('Invalid Repository url');
+
+		branch = await listSelection(
+			branches,
+			'Select a container to get logs'
```

Comment:
Change the text, you are asking for logs selection but you want the user to select a specific branch